core/unit_test/TestViewPadding.hpp

```diff
@@ -0,0 +1,618 @@
+  return Kokkos::subview(view, pair, Kokkos::ALL);
+}
+
+template <typename dim_t, typename layout, typename viewDim>
+auto get_subview(
+    Kokkos::View<dim_t, layout> &view,
+    [[maybe_unused]] typename std::enable_if_t<
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_3>>::value, void>
+        *ptr = nullptr) {
+  auto pair = pair_t(SUBRANGE_DIM);
+  return Kokkos::subview(view, pair, Kokkos::ALL, Kokkos::ALL);
+}
+
+template <typename dim_t, typename layout, typename viewDim>
+auto get_subview(
+    Kokkos::View<dim_t, layout> &view,
+    [[maybe_unused]] typename std::enable_if_t<
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_4>>::value, void>
+        *ptr = nullptr) {
+  auto pair = pair_t(SUBRANGE_DIM);
+  return Kokkos::subview(view, pair, Kokkos::ALL, Kokkos::ALL, Kokkos::ALL);
+}
+
+template <typename dim_t, typename layout, typename viewDim>
+auto get_subview(
+    Kokkos::View<dim_t, layout> &view,
+    [[maybe_unused]] typename std::enable_if_t<
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_5>>::value, void>
+        *ptr = nullptr) {
+  auto pair = pair_t(SUBRANGE_DIM);
+  return Kokkos::subview(view, pair, Kokkos::ALL, Kokkos::ALL, Kokkos::ALL,
+                         Kokkos::ALL);
+}
+
+template <typename dim_t, typename layout, typename viewDim>
+auto get_subview(
+    Kokkos::View<dim_t, layout> &view,
+    [[maybe_unused]] typename std::enable_if_t<
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_6>>::value, void>
+        *ptr = nullptr) {
+  auto pair = pair_t(SUBRANGE_DIM);
+  return Kokkos::subview(view, pair, Kokkos::ALL, Kokkos::ALL, Kokkos::ALL,
+                         Kokkos::ALL, Kokkos::ALL);
+}
+
+template <typename dim_t, typename layout, typename viewDim>
+auto get_subview(
+    Kokkos::View<dim_t, layout> &view,
+    [[maybe_unused]] typename std::enable_if_t<
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_7>>::value, void>
+        *ptr = nullptr) {
+  auto pair = pair_t(SUBRANGE_DIM);
+  return Kokkos::subview(view, pair, Kokkos::ALL, Kokkos::ALL, Kokkos::ALL,
+                         Kokkos::ALL, Kokkos::ALL, Kokkos::ALL);
+}
+
+template <typename dim_t, typename viewDim, typename layout,
+          typename allow_padding, typename... Vals>
+typename std::enable_if_t<
+    std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_0>>::value ||
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_1>>::value,
+    bool>
+padding_subview_and_copy(int rank, Vals... params) {
+  bool result  = true;
+  using view_t = Kokkos::View<dim_t, layout>;
+  auto alloc_prop =
+      Kokkos::view_alloc("vDim" + std::to_string(rank), Kokkos::AllowPadding);
+  view_t v(alloc_prop, params...);
+  auto v_sub        = get_subview<dim_t, layout, viewDim>(v);
+  auto v_sub_mirror = Kokkos::create_mirror_view(v_sub);
+  result &= ASSERT_VIEW_STRIDES(v, v_sub, rank);
+  /* We never pad for rank 0 and rank 1 and for LR && dyn_rank > 1 */
```

Comment:
The comment mentions `LayoutRight && dyn_rank > 1`, but the code doesn't make such a distinction - it only considers rank.